case-lib/lib.sh

```diff
@@ -1,11 +1,13 @@
 #!/bin/bash
 
+declare -g SCRIPT_HOME=$(realpath "$(dirname $0)/../")
+
```

Comment:
1. You quoted "$( )" which is not necessary. You didn't quote "$0" which _is_ necessary in case of whitespace and reported by shellcheck.

2. I don't think this `realpath` solves any issue (does it?) and it could create some confusion for users with a symbolic link (/home -> /nfs/something or other)

Here's a tested way to make SCRIPT_HOME absolute and whitespace-proof without resolving symbolic links:
```shell
SCRIPT_HOME=$(cd $(dirname "$0")/..; pwd)
```

3. The `-g` has no effect outside functions, did you mean `export`?

Reply:
yes, you are right  
I try the `dirname $(dirname $0)/../` with `realpath $(dirname $0)../` and `cd $(dirname "$0")/..; pwd`   
the `realpath` maybe confuse when dump the `$SCRIPT_HOME`  
for the option 3, it just my person style, I more like to told the reader to notice this value will be used at other place